tests/mock_singleton_helpers.h

```diff
@@ -26,27 +26,39 @@
 #include <utility>
 
 #define MP_MOCK_SINGLETON_INSTANCE(mock_class)                                                                         \
+public:                                                                                                                \
     static mock_class& mock_instance()                                                                                 \
     {                                                                                                                  \
         return dynamic_cast<mock_class&>(instance());                                                                  \
     }
 
+#define MP_MOCK_SINGLETON_HELPER_TYPES(mock_class, parent_class)                                                       \
+private:                                                                                                               \
+    static constexpr auto academy = [] { return sg::make_scope_guard([] { parent_class::reset(); }); }; /*             \
```

Comment:
Interesting name for a lambda.  Just curious how you arrived at this name :smile:

Reply:
Hehe, I just wanted the name of the entity that produces "guards" and that was the best I could think of. I also thought a little humor wouldn't hurt :smile: I had "barracks" before even. I also thought of "squad" or "squadron" but went with this.

However, if it's unclear I'm ofc fine with changing. Would you have any suggestions?

Reply:
Ah, I see, like a military academy.  Very clever :smiley:  I think the name is fine given that I'm clued into the humor, but for other readers who may not "get it", perhaps just a little comment above saying something like `Produces "guards"` would help?

Reply:
Sure, I will add that.

Reply:
Thanks!